### ISSUE TITLE
Extension libraries reference dependencies not found in NuGet feeds

### ISSUE THREAD

**Issue by WalternativE** [dotnet/interactive#852]
### Describe the bug
First of all: Thanks for your great work. Have been using dotnet interactive a lot since the VSCode integration was released and it is a joy to work with ❤❤❤

I just found a little thing. Referencing the packages meant to build extensions as documented in the main Readme (eg. `Microsoft.DotNet.Interactive`, `Microsoft.DotNet.Interactive.Formatting`, etc) leads to failing project restores.

Dependencies like `System.CommandLine` can be fixed when replicating the `NuGet.config` used in the dotnet interactive repo. Dependencies like `FSAutocomplete.Core` cannot be fixed as they do not exist in any feeds but only as project references.

I created a repository with two configurations [the main branch](https://github.com/WalternativE/SillyKernelExtension) has a working version of the plugin using your `NuGet.config`. The [broken-references branch](https://github.com/WalternativE/SillyKernelExtension/tree/broken-references) uses the global standard. If you want to test this **you first have to clean your nuget cache** because otherwise it seems as if it would work nicely (which it doesn't).

I'm working on another extension where the problem is a bit more severe. [This Deedle formatter extension](https://github.com/WalternativE/Deedle.DotNet.Interactive.Extension) is currently blocked from upgrading the packages because I can't find a workaround for dependencies like `FSAutocomplete.Core`.

I'm totally fine with using the custom `NuGet.config`. I just think it should be reflected in the documentation. People using packages from nuget.org are kind of used to having it as the only repo - even for prerelease packages.

### Please complete the following:

**Which version of .NET Interactive are you using?** There are a few ways to find this out:

Version: 1.0.151601+c44fed791093db19261c49b39d7d901e7983a3af
Build date: 2020-10-16T22:22:16.0000000Z
This doesn't really connect to my issue though. My pinned .NET Core SDK version is 3.1.403

- OS
    - [x] Windows 10 (working on WSL2 - Ubuntu 20.04)
    - [ ] macOS
    - [ ] Linux (Please specify distro)
    - [ ] iOS
    - [ ] Android
 - Browser 
    - [ ] Chrome
    - [ ] Edge
    - [ ] Firefox
    - [ ] Safari
    - [x] Does not apply
 - Frontend
    - [ ] Jupyter Notebook
    - [ ] Jupyter Lab
    - [ ] nteract
    - [x] Visual Studio Code
    - [ ] Other (please specify)

### Screenshots
![image](https://user-images.githubusercontent.com/5071748/96913353-cb6bda80-14a3-11eb-8b41-391be56bfe38.png)

**Comment by jonsequitur** [comment]
The FsAutoComplete.Core package dependency has been removed. Does this unblock you?

We'll update to and publish a newer System.CommandLine next.

**Comment by WalternativE** [comment]
I would think so. Is there a NuGet feed I can use to test this? I imagine this change won't make it to the 'standard' NuGet feed for a while, right?

**Comment by jonsequitur** [comment]
Our build pipeline publishes packages here on every merge:

https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json

**Comment by WalternativE** [comment]
I upgraded to the newest version of the packages in the feed you mentioned (as of writing this would be `1.0.0-beta.20522.4`). I cleaned my NuGet cache and it restored/built without problems. I'd say that using those feeds I'd be unblocked 👍

I'd still document the need for those feeds somewhere for extension authors. Maybe it's just me but when I see a package on the main NuGet feed I'd usually think that all its dependencies are also on the main feed. I could be totally wrong here as well, though.

**Comment by jonsequitur** [comment]
The goal is that all of the dependencies are on public NuGet going forward as we improve the extension story and docs. We're trimming down the dependencies list and publishing an updated System.CommandLine soon. 

Thanks for being an early adopter!

**Comment by WalternativE** [comment]
I understand. Thanks for all your work on the extension mechanism! Really nice to have a way to customize outputs to our (maybe sometimes fringe - speaking as a F# dev) needs 🙇‍♂️

**Comment by Alkasai** [comment]
Dumb question - so how do I get this extension in .NET Interactive Notebook?

**Comment by jonsequitur** [comment]
@Alkasai: https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.dotnet-interactive-vscode

**Comment by Alkasai** [comment]
@jonsequitur I meant the Deedle Formatter extension.

**Comment by WalternativE** [comment]
@Alkasai it's not part of this repo (or Deedle for that matter because it's currently one of my pet projects). You can find the repo here https://github.com/WalternativE/Deedle.DotNet.Interactive.Extension - there is also a short description on how to get it because it's currently hosted on MyGet. If you have problems with it let me know (best would be in the repo). Hope this answers your question.
